licenses/management/commands/publish.py

```diff
@@ -53,35 +52,53 @@ def add_arguments(self, parser: ArgumentParser):
             action="store_true",
             help="A list of active branches in cc-licenses-data will be displayed",
         )
+        parser.add_argument(
+            "--nopush", action="store_true", help="Do not push upstream",
+        )
 
     def _quiet(self, *args, **kwargs):
         pass
 
     def run_django_distill(self):
         """Outputs static files into the specified directory determined by settings.base.DISTILL_DIR
         """
-        stdout = self._quiet
         output_dir = getattr(settings, "DISTILL_DIR", None)
         if not os.path.isdir(settings.STATIC_ROOT):
             e = "Static source directory does not exist, run collectstatic"
             raise CommandError(e)
         output_dir = os.path.abspath(os.path.expanduser(output_dir))
+        print(f"output_dir={output_dir}. Will delete, then generate HTML files there.")
         if os.path.isdir(output_dir):
             rmtree(output_dir)
         os.makedirs(output_dir)
-        try:
-            render_to_dir(output_dir, urls_to_distill, stdout)
-        except DistillError as err:
-            raise CommandError(str(err)) from err
+
+        for legalcode in LegalCode.valid():
+            save_url_as_static_file(output_dir, legalcode.license_url())
+            save_url_as_static_file(output_dir, legalcode.deed_url())
+        save_url_as_static_file(output_dir, "/")
+        save_url_as_static_file(output_dir, "/status/")
+        for tbranch in TranslationBranch.objects.filter(complete=False).only("id"):
+            save_url_as_static_file(output_dir, f"/status/{tbranch.id}/")
 
     def publish_branch(self, branch: str):
         """Workflow for publishing a single branch"""
+        print(f"Publishing branch {branch}")
         with git.Repo(settings.TRANSLATION_REPOSITORY_DIRECTORY) as repo:
+            if repo.is_dirty():
+                raise Exception("Repo is dirty, will not publish")
```

Comment:
Instead of "is dirty" can we include a more informative response for the Exception.

Reply:
will do.